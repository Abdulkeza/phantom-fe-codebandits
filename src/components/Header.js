import React, { forwardRef, useState } from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import phantom from '../assets/phantom.png';
import xmark from '../assets/xmark.svg';
import Burger from './Burger.js';
import Button from './Button.js';
import LanguageButton from './LanguageButton.js';
import MobileMenu from './MobileMenu.js';
import withClickOutside from './WithClickOutside.js';

const Header = forwardRef(({ open, setOpen }, ref) => {
  const { authenticated, user } = useSelector((state) => state?.auth);
  const [style, setStyle] = useState('hidden');
  const [icon, setIcon] = useState(<Burger />);
  const handleToggleMenu = () => {
    if (style === 'hidden') {
      setStyle('flex z-50');
      setIcon(<img src={xmark} alt="x mark" className="w-6" />);
    } else {
      setStyle('hidden');
      setIcon(<Burger />);
    }
  };
  return (
    <div className="flex bg-background border-b-4 pb-4 sticky top-0 z-20">
      <div>
        <Link to="/">
          <img
            src={phantom}
            alt="Phantom"
            className="w-44 xl:w-60 ml-8 xl:ml-20 mt-7 mb-2"
          />
        </Link>
      </div>
      <div className="hidden md:flex xl:flex ml-auto mr-20">
        <Link to={!authenticated ? '/login' : '/dashboard'}>
          <Button
            name={!authenticated ? 'For operators' : 'Dashboard'}
            styles="bg-primary hover:bg-hover text-white rounded-xl mt-7 ml-6"
          />
        </Link>
        <Link to="/track">
          <Button
            name="Track Bus"
            styles="border border-primary bg-background hover:bg-hover2 text-primary rounded-xl mt-7 ml-6 px-6"
          />
        </Link>

        <LanguageButton styles="mt-8" />
      </div>
      <section
        ref={ref}
        className="flex flex-col absolute right-0 bg-background px-2"
      >
        <button
          type="button"
          id="toggle"
          className="md:hidden xl:hidden ml-auto mr-12 mt-8"
          onClick={() => setOpen(!open)}
        >
          {!open && <Burger />}
          {open && <img src={xmark} alt="xmark" className="w-6 mb-2" />}
        </button>

        {open && <MobileMenu />}
      </section>
    </div>
  );
});

export default withClickOutside(Header);
