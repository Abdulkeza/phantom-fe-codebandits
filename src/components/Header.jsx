import React from 'react';
import Button from './Button.jsx';
import phantom from '../assets/phantom.png';

const Header = () => {
  return (
    <div className="flex border-b-4 pb-4">
      <img src={phantom} alt="Phantom" className="w-60 ml-20 mt-10" />
      <div className="flex ml-auto mr-20">
        <Button
          name="For operators"
          styles="bg-blue-600 text-white rounded-xl mt-10 ml-6"
        />
        <Button
          name="Track Bus"
          styles="border border-blue-600 bg-white text-blue-600 rounded-xl mt-10 ml-6 px-6"
        />
        <h2 className="mt-12 ml-4 py-2 px-4">Language</h2>
      </div>
    </div>
  );
};

export default Header;
