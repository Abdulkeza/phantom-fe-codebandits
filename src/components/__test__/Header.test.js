import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../Header.js';

describe('Header', () => {
  it('should render Header', () => {
    const elem = renderer.create(<Header />).toJSON();
    expect(elem).toMatchSnapshot();
  });
});
