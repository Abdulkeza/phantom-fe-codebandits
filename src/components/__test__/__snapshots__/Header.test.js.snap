// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Header 1`] = `
<div
  className="flex bg-background border-b-4 pb-4 sticky top-0 z-20"
>
  <div>
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Phantom"
        className="w-44 xl:w-60 ml-8 xl:ml-20 mt-7 mb-2"
        src={null}
      />
    </a>
  </div>
  <div
    className="hidden md:flex xl:flex ml-auto mr-20"
  >
    <a
      href="/login"
      onClick={[Function]}
    >
      <div>
        <button
          className="bg-primary hover:bg-hover text-white rounded-xl mt-7 ml-6 rounded-xl py-3 px-5"
          type="button"
        >
          For operators
        </button>
      </div>
    </a>
    <div>
      <button
        className="border border-primary bg-background hover:bg-hover2 text-primary rounded-xl mt-7 ml-6 px-6 rounded-xl py-3 px-5"
        type="button"
      >
        Track Bus
      </button>
    </div>
    <div>
      <button
        className="undefined md:mt-8 lg:mt-8 xl:mt-8 flex py-2 px-4 text-lg hover:font-bold"
        onClick={[Function]}
        type="button"
      >
        <h1>
          English
        </h1>
        <img
          alt="dropdown"
          className="w-3 mt-3 ml-2 h-2"
          src={null}
        />
      </button>
    </div>
  </div>
  <section
    className="flex flex-col absolute right-0 bg-background px-2"
  >
    <button
      className="md:hidden xl:hidden ml-auto mr-12 mt-8"
      onClick={[Function]}
      type="button"
    >
      <div>
        <div
          className="h-1 w-7 bg-primary mt-1"
        />
        <div
          className="h-1 w-7 bg-primary mt-1"
        />
        <div
          className="h-1 w-7 bg-primary mt-1"
        />
      </div>
    </button>
  </section>
</div>
`;
